testsuite/tests/quotation/build_quotation.ml

```diff
@@ -686,23 +693,56 @@ let x = <[ 123 ]> in <[ $x ]>;;
 - : <[int]> expr = <[123]>
 |}];;
 
+let _ = <[ [ a * b for a = 1 to 10 for b = a to 10 ] ]>;;
+[%%expect {|
+- : <[int list]> expr = <[[ a * b for a = 1 to 10 for b = a to 10 ]]>
+|}];;
+
+let _ = <[ [ a * b for a = 1 to 10 and b = 1 to 10 ] ]>;;
+[%%expect {|
+- : <[int list]> expr = <[[ a * b for b = 1 to 10 for a = 1 to 10 ]]>
```

Comment:
This translation of comprehensions is semantically incorrect according to https://beta.oxcaml.org/documentation/comprehensions/01-intro/. I'd massively prefer if the quote representation of comprehensions was much closer to that of parsetree and typedtree. (The printer for comprehensions also makes me unhappy).

Reply:
This has not been fixed.